### PR TITLE
Feat/metadata improvements

### PR DESCRIPTION
**Pull request by Ryanmtate**
Adds Client Metadata accessor methods according to:

- https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-5.1-4.2.1
- https://openid.net/specs/oauth-v2-jarm-final.html#name-client-metadata

### CODE REVIEW

File: src/core/authorization_request/parameters.rs [hunk 1]
```diff
@@ -169,6 +188,150 @@ impl ClientMetadata {
...
+
+    /// Return the `VpFormats` from the `client_metadata` field.
+    ///
+    /// vp_formats: REQUIRED when not available to the Wallet via another mechanism.
+    ///
+    /// As defined in [Section 10.1](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#client_metadata_parameters).
+    ///
+    /// See reference: https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-5.1-4.2.2.2
+    pub fn vp_formats(&self) -> Result<VpFormats, Error> {
+        self.0.get().ok_or(anyhow!("missing vp_formats"))?
```
Comment (cobward) [inline src/core/authorization_request/parameters.rs:224]:
```suggestion
        self.0.get().parsing_error()?
```

File: src/core/authorization_request/parameters.rs [hunk 2]
```diff
@@ -169,6 +188,150 @@ impl ClientMetadata {
...
+    ///
+    /// The jwks_uri or jwks metadata parameters can be used by clients to register their public encryption keys.
+    ///
+    /// See: https://openid.net/specs/oauth-v2-jarm-final.html#section-3-4
+    ///
+    pub fn jwks(&self) -> Option<Vec<JWK>> {
+        self.0 .0.get("jwks").and_then(|j| j.as_array()).map(|j| {
+            j.iter()
+                // NOTE: we're ignoring the errors here, but may wish to handle the errors in the future.
+                .filter_map(|jwk| serde_json::from_value(jwk.clone()).ok())
```
Comment (cobward) [inline src/core/authorization_request/parameters.rs]: This and the other parameters that you have manually implemented are already implemented here:
https://github.com/spruceid/openid4vp/blob/main/src/core/metadata/parameters/verifier.rs

You should change these methods to `self.0.get()` as for vp_formats, or just do `client_metadata.0.get()` in the downstream code.

**Review by cobward — CHANGES_REQUESTED** [review]

File: src/core/authorization_request/parameters.rs (hunk 2 above)
Comment (Ryanmtate) [reply src/core/authorization_request/parameters.rs]: Thanks for pointing this out. I updated to use the typed parameter implementations, and added missing typed parameter implementations for the other fields.

**Review by cobward — APPROVED** [review]

File: src/core/metadata/parameters/wallet.rs [hunk 3]
```diff
@@ -205,6 +205,73 @@ impl From<AuthorizationEncryptionEncValuesSupported> for Json {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct AuthorizationSignedResponseAlg(pub ssi::crypto::Algorithm);
+
```
Comment (cobward) [inline src/core/metadata/parameters/wallet.rs]: These new values should be in `verifier.rs` not `wallet.rs`, right?

**Review by cobward — CHANGES_REQUESTED** [review]
One final comment, otherwise LGTM

File: src/core/metadata/parameters/wallet.rs (hunk 3 above)
Comment (Ryanmtate) [reply src/core/metadata/parameters/wallet.rs]: Nice call, and I realize you already had some of these implemented.

I've removed the extraneous implementations in [204fc40](https://github.com/spruceid/openid4vp/pull/43/commits/204fc40606f0b2389fdf728d885d1649f406b58b)

**Review by cobward — APPROVED** [review]

**Pull request merged**